activemodel/lib/active_model/attribute_methods_builder.rb

```diff
@@ -0,0 +1,265 @@
+# frozen_string_literal: true
+
+require "concurrent/map"
+
+module ActiveModel
+  # == Active \Model \Attribute \Methods\ Builder
+  #
+  # Module builder to define attribute methods on a class. Used by
+  # <tt>ActiveModel::AttributeMethods</tt> to add prefixes and suffixes to
+  # models.
+  #
+  # The builder can also be used on its own:
+  #
+  #   class Person
+  #     mod = ActiveModel::AttributeMethodsBuilder.new
+  #
+  #     include mod
+  #
+  #     mod.affix prefix: 'reset_', suffix: '_to_default!'
+  #     mod.suffix '_contrived?'
+  #     mod.prefix 'clear_'
+  #     mod.define_attribute_methods :name
+  #
+  #     attr_accessor :name
+  #
+  #     def attributes
+  #       { 'name' => @name }
+  #     end
+  #
+  #     private
+  #
+  #     def attribute_contrived?(attr)
+  #       true
+  #     end
+  #
+  #     def clear_attribute(attr)
+  #       send("#{attr}=", nil)
+  #     end
+  #
+  #     def reset_attribute_to_default!(attr)
+  #       send("#{attr}=", 'Default Name')
+  #     end
+  #   end
+  #
+  class AttributeMethodsBuilder < Module
```

Comment:
I don't think we should make this module public API. The public API is `ActiveModel::AttributeMethods`

Reply:
Fair enough. Should I snug the builders into the same files as AM::AttributeMethods / AR:AttributeMethods, or can I just remove the docs and put a `# :nodoc:` in there?